client/src/providers/UserProvider.tsx

```diff
@@ -56,5 +61,9 @@ export function UserProvider(props: React.PropsWithChildren<{}>): JSX.Element {
     setUserTimeout(null, true, true);
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  return <UserContext.Provider value={[user, setUserTimeout]}>{children}</UserContext.Provider>;
+  return (
+    <UserContext.Provider value={[user, setUserTimeout, isLoading]}>
+      {children}
+    </UserContext.Provider>
+  );
```

Comment:
added loading state so that we can handle with `if isLoading return null` in our components and therefore always have `user` var in renders and not have to deal with conditionals